src/main/java/wellnus/storage/ReflectionTokenizer.java

```diff
@@ -0,0 +1,110 @@
+        }
+        String tokenizedIndexes = tokenizedRawString.substring(i).trim();
+        return tokenizedIndexes;
+    }
+    private String[] splitTokenizedIndex(String tokenizedIndexes) {
+        tokenizedIndexes = tokenizedIndexes.strip();
+        int noLimit = -1;
+        String[] rawString = tokenizedIndexes.split(INDEX_DELIMITER, noLimit);
+        String[] cleanString = new String[rawString.length];
+        for (int i = 0; i < rawString.length; ++i) {
+            String currentCommand = rawString[i];
+            currentCommand = currentCommand.strip();
+            cleanString[i] = currentCommand;
+        }
+        return cleanString;
+    }
+    private Set<Integer> getSet(String indexToSplit) throws TokenizerException {
+        Set<Integer> cleanIndexes = new HashSet<>();
+        if (indexToSplit.isBlank()) {
+            return cleanIndexes;
+        }
+        String[] splittedString = splitTokenizedIndex(indexToSplit);
+        try {
+            for (String indexString: splittedString) {
+                int index = Integer.parseInt(indexString);
+                cleanIndexes.add(index);
+            }
+        } catch (NumberFormatException numberFormatException) {
+            throw new TokenizerException(DETOKENIZE_ERROR_MESSAGE);
+        }
+        return cleanIndexes;
+    }
+    /**
+     * Tokenize ArrayList of Set of Integers into strings that can be stored. <br>
+     * ArrayList contains 2 Set of Integers, which corresponds for set of like indexes for the first entry
+     *      and set of pref indexes for second entry.<br>
+     * Each index will be tokenized with the following format:
+     * like [list of comma separated index] <br>
+     * pref [list of comma separated index] <br>
+     *
+     * @param arrayIndexToTokenize ArrayList that contains set of like indexes for the first entry
+     *      and set of pref indexes for the second entry. <br>
+     * @return ArrayList of Strings representing the tokenized like indexes and pref indexes that we can
+     *      write to storage.
+     */
+    public ArrayList<String> tokenize(ArrayList<Set<Integer>> arrayIndexToTokenize) throws TokenizerException {
+        ArrayList<String> tokenizedIndex = new ArrayList<>();
+        Set<Integer> likeIndexToTokenize = arrayIndexToTokenize.get(INDEX_ZERO);
+        Set<Integer> prefIndexToTokenize = arrayIndexToTokenize.get(INDEX_ONE);
+        String tokenizedLike = LIKE_KEY + " ";
+        for (int indexLike: likeIndexToTokenize) {
+            tokenizedLike = tokenizedLike + indexLike + INDEX_DELIMITER;
+        }
+        tokenizedLike = tokenizedLike.substring(INDEX_ZERO, tokenizedLike.length() - INDEX_ONE);
+        String tokenizedPref = PREF_KEY + " ";
+        for (int indexPref: prefIndexToTokenize) {
+            tokenizedPref = tokenizedPref + indexPref + INDEX_DELIMITER;
+        }
+        tokenizedPref = tokenizedPref.substring(INDEX_ZERO, tokenizedPref.length() - INDEX_ONE);
```

Comment:
Hmm this chunk of code is the same logic as the previous 4 lines, I would recommend abstracting it out as a `private` method to shorten and simplify your code and reduce repetition(bad for maintainence)